src/bridge/cockpitpeer.c

```diff
@@ -279,16 +335,18 @@ on_other_control (CockpitTransport *transport,
         g_source_remove (self->authorize_values_timeout);
       self->authorize_values_timeout = g_timeout_add (2*60*1000, cockpit_peer_delete_authorize_values, self);
 
-      startup_done (self, problem);
-
       if (problem)
         {
           cockpit_transport_close (transport, problem);
+          startup_drain_stderr (self);
```

Comment:
This doesn't drain everything... we should call g_io_channel_read_to_end, I guess.

Reply:
I put a loop around startup_stderr_drain.